### ISSUE TITLE
Start a Report: As a developer, I want an API endpoint to  "Start" and "Continue" a Report

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-reporting#248]
#### Description:

This card was split out of #229. Please refer to the contents of that card for further context.

As an Industry Reporter, I want to click on "Start" in the Action column for a specific Operation listed on the Reporting Dashboard so that I can start a report for that Operation, for the current reporting year.

As an Industry Reporter, I want to click on "Continue" in the action column for a specific operation where I've already started a report.

~blocked by #249~

#### Acceptance Criteria:

Given I am on my operator's operation table
When I haven't started a report for that operation
And I click on "Start" in the Action column
Then a new report is created in the database
And it is seeded with Operator/Operation/Facility data from registration

Given I am on my operator's operation table
When I have previously started a report for that operation
And I click on "Continue" in the Action column
Then I am directed to the first page of the in-progress report

#### Development Checklist:

- [ ] ~Start link in the Action column of the operation table~ #274 
- [x] New record in Report table with appropriate data
- [x] New record in report_facility and report_operation data, with data from the registration table
- [ ] ~Continue link in the Action column in the operation table that directs to the in-progress report~ #274 
- [x] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by pbastia** [comment]
Card is unblocked now that  #249 is done

**Comment by patriciarussellCAS** [comment]
This is the API ticket - user story is captured in #274
